### DISCUSSION TITLE
FastAPI Performance

### DISCUSSION THREAD

**Discussion by wuluis** [fastapi/fastapi#7049] [Questions] [answered]
The README.md file for FastAPI being on par seems misleading I think they could say that it's one of the fastest python frameworks to be more precise since I'm now going to edit this what I'm saying is FastAPI can't handle that many requests per second.

**Comment by wuluis** [comment]
Is there any way to speed up the request per second?

**Comment by includeamin** [comment]
how do you run FastApi with uvicorn or gunicorn?
what is your gunicorn config?
`requests per second` depend on many parameters.

**Comment by wuluis** [comment]
I'm just doing this "uvicorn main:app --reload"

**Comment by wuluis** [comment]
I used autocannon for the test "autocannon -c 100 -d 40 -p 10 localhost:8000", 2 times the first one was for a warm up. If I'm doing this wrong then my apologies I'm new to using a framework with python.

**Comment by includeamin** [comment]
[check it out](https://www.uvicorn.org/#running-with-gunicorn)

**Comment by wuluis** [comment]
Okay

**Comment by wuluis** [comment]
Oh it's because I'm using windows

**Comment by ycd** [comment]
What made you think the problem is related to your OS?

**Comment by wuluis** [comment]
Google

**Comment by wuluis** [comment]
@ycd Do you have any test which could show me the req per sec you got?

**Comment by ycd** [comment]
There are a lot of tools out there, [ali](https://github.com/nakabonne/ali), [hey](https://github.com/rakyll/hey), and well-rounded benchmarks like TechEmpowers.

**Comment by wuluis** [comment]
Okay thanks.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the help here everyone! :clap: :bow: 

---

@mcbrowny35 "on par with" means:

> at the same level or standard as (someone or something else)

Ref: https://www.merriam-webster.com/dictionary/on%20par%20with

"on par with Go" or "on par with Node.js" would mean that it is comparable to those languages. And those languages include a wide variety of frameworks. It doesn't mean "faster than every other framework made with any of those languages". In that case, the phrase would have been "faster than Go or Node.js".

And as you can see on the benchmarks, it can be a bit faster than _some_ of the frameworks in those languages.

Now, if you want to improve performance and deploy to production, you shouldn't use `--reload`, that exclusively for development. And you shouldn't deploy to Windows. 90% of the cloud servers run on Linux. You should do that as well.
